main.py

```diff
@@ -64,6 +64,39 @@
 PUNISHMENT_INACTIVITY = 5
 
 
+def calculate_all_fitness(end_round_time, player_list, genomes_list):
+    if time.time() >= end_round_time or len(player_list) < 2:
+            genomes_list = list(map(calculate_fitness, player_list))
+            #for player_index, player in enumerate(player_list):
+            #    genomes_list[player_index].fitness = calculate_fitness(player)
+            print("Round time done")
+            for i in reversed(range(3)):
+                print(f"Starting next generation in {i+1}...", end="\r")
+                time.sleep(1)
+            return True
+    return False
+
+def correct_conflicting_movements(keys):
+    keys[pygame.K_LEFT] = keys[pygame.K_LEFT] ^ keys[pygame.K_RIGHT]
+    keys[pygame.K_DOWN] = keys[pygame.K_DOWN] ^ keys[pygame.K_UP]
+    keys[pygame.K_UP] = keys[pygame.K_DOWN] ^ keys[pygame.K_UP]
+    keys[pygame.K_RIGHT] = keys[pygame.K_DOWN] ^ keys[pygame.K_UP]
+
+def remove_element_from_lists( index, *args):
+    if type(index) != int:
+        print("Ivalid type of index arg")
+    for arg in args:
+        arg.pop(index)
+    
+def check_player_collision(players_list, player):

```

Comment:
No idea if this is just the in-browser window bugging out but indentation of the logic in that for loop looks over-indented.

Reply:
Just the browser I think